### PR TITLE
Fix installation from npm

### PR DESCRIPTION
**Pull request by nosemeocurrenada**
+ Moved the command,args choice to a getCommand function

This function was added only for the sake of clarity

+ Add files to the package.json
        
The npm installation was failing because it couldn't find the needed files.
Now index.js is a file that only exports the service function. And app.js is the actual app that runs.
To test the npm installation, you may run `npm pack` in the wetty directory, it will generate a file like `wetty.js-0.4.0.tgz`. Then, given that you cd'ed to another node project, `npm install path/to/wetty.js-0.4.0.tgz` should install it in the node_modules directory

### CODE REVIEW

File: wetty.js [hunk 1]
```diff
@@ -36,21 +36,32 @@ function createServer(port, sslopts) {
...
+    cmd = '/usr/bin/login';
+    args = ['-h', socket.client.conn.remoteAddress.split(':')[3]];
+  } else {
+    cmd = path.join(__dirname, 'bin/ssh');
+    args = [ssh, '-p', sshport, '-o', `PreferredAuthentications=${sshauth}`];
+  }
+  return [cmd, args, ssh];
+}
+
+module.exports = function start(port, sshuser, sshhost, sshport, sshauth, sslopts) {
```
Comment (butlerx) [inline wetty.js]: can you use js [export](https://developer.mozilla.org/en/docs/web/javascript/reference/statements/export) rather then module.exports

File: wetty.js [hunk 2]
```diff
@@ -36,21 +36,32 @@ function createServer(port, sslopts) {
...
-    const match = request.headers.referer.match('.+/ssh/.+$');
-    const sshAddress = globalsshuser ? `${globalsshuser}@${sshhost}` : sshhost;
-    const ssh = match ? `${match[0].split('/ssh/').pop()}@${sshhost}` : sshAddress;
-    const cmd = process.getuid() === 0 && sshhost === 'localhost' ? '/usr/bin/login' : './bin/ssh';
-    const args =
-      cmd === './bin/ssh'
-        ? [ssh, '-p', sshport, '-o', `PreferredAuthentications=${sshauth}`]
-        : ['-h', socket.client.conn.remoteAddress.split(':')[3]];
+
+    const [cmd, args, ssh] = getCommand(socket, sshuser, sshhost, sshport, sshauth);
```
Comment (butlerx) [inline wetty.js]: should this not be globalsshuser?

File: package.json [hunk 3]
```diff
@@ -54,17 +54,20 @@
   },
   "homepage": "https://github.com/butlerx/wetty",
   "preferGlobal": "true",
-  "main": "app.js",
+  "main": "index.js",
   "bin": {
-    "wetty": "./app.js"
+    "wetty": "./index.js"
```
Comment (butlerx) [inline package.json]: should this not still be app.js

File: app.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
 #! /usr/bin/env node
 require('@std/esm');
 require('./cli');
-module.exports = require('./wetty').default;
+module.exports = require('./wetty');
```
Comment (butlerx) [inline app.js]: if this file is just for running does it still need module.exports

File: wetty.js (hunk 2 above)
Comment (nosemeocurrenada) [reply wetty.js]: I've changed it to sshuser to keep the names similar. Do you prefer it as globalsshuser?

Comment (butlerx) [reply wetty.js]: oh id not noticed youd changed the args my mistake

**Pull request merged**